src/credentials/loginManager.ts

```diff
@@ -69,15 +69,13 @@ export class LoginManager {
         } catch (err) {
             // TODO: don't hardcode logic using error message, have a 'type' field instead
             if (!(err as Error).message.includes('cancel')) {
```

Comment:
I think the relevant PR was merged for this

```suggestion
            if (!(err instanceof TimeoutError && err.type === 'cancelled')) {
```

Could make this predicate a static method of the `TimeoutError` class as well:
```ts
public static isCancelled(err: any): err is TimeoutError & { type: 'cancelled' } {
  return err instanceof TimeoutError && err.type === 'cancelled'
}
```
Still trying to figure out a better name for `Timeout`. It's just a cancellation token with an expiration built-in. So I guess `CancellationToken`, but VS Code already has an interface for this. At some point this should just be consolidated on top of VS Code's own cancellation API.

Reply:
I will make this change in a fast-follow PR (which also adds another feature).

> At some point this should just be consolidated on top of VS Code's own cancellation API.

Until then, I think "Timeout" is probably fine. "Timeout" is kind of a primitive in our codebase by now.